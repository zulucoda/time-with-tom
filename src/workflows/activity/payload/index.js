import activityConfig from '../constants';

/**
 * map payload for workflow
 * @function
 * @param {object} values - formik values
 * @returns {object} mappedPayload - returns mapped payload
 */
const payload = (values) => {
  return {
    [activityConfig.SELECT_NAME]: values[activityConfig.SELECT_NAME],
  };
};

export default payload;
