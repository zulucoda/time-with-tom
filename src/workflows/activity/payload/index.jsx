const getPayload = (values) => {
  return { ...values };
};

export default getPayload;
