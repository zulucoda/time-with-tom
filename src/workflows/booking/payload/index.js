import bookingConfig from '../constants';

/**
 * map payload for workflow
 * @function
 * @param {object} values - formik values
 * @returns {object} mappedPayload - returns mapped payload
 */
const payload = (values) => {
  return {
    [bookingConfig.INPUT_DATE]: values[bookingConfig.INPUT_DATE],
    [bookingConfig.INPUT_START_TIME]: values[bookingConfig.INPUT_START_TIME],
    [bookingConfig.INPUT_END_TIME]: values[bookingConfig.INPUT_END_TIME],
  };
};

export default payload;
