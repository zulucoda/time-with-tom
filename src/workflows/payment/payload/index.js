import paymentConfig from '../constants';

/**
 * map payload for workflow
 * @function
 * @param {object} values - formik values
 * @returns {object} mappedPayload - returns mapped payload
 */
const payload = (values) => {
  return {
    [paymentConfig.INPUT_CC_CVC]: values[paymentConfig.INPUT_CC_CVC],
    [paymentConfig.INPUT_CC_EXPIRY_YEAR]:
      values[paymentConfig.INPUT_CC_EXPIRY_YEAR],
    [paymentConfig.INPUT_CC_EXPIRY_MONTH]:
      values[paymentConfig.INPUT_CC_EXPIRY_MONTH],
    [paymentConfig.INPUT_CC_NUMBER]: values[paymentConfig.INPUT_CC_NUMBER],
    [paymentConfig.INPUT_CC_NAME]: values[paymentConfig.INPUT_CC_NAME],
  };
};

export default payload;
