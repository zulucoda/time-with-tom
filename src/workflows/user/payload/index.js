import userConfig from '../constants';

/**
 * map payload for workflow
 * @function
 * @param {object} values - formik values
 * @returns {object} mappedPayload - returns mapped payload
 */
const payload = (values) => {
  return {
    [userConfig.INPUT_NAME]: values[userConfig.INPUT_NAME],
    [userConfig.INPUT_SURNAME]: values[userConfig.INPUT_SURNAME],
    [userConfig.INPUT_EMAIL]: values[userConfig.INPUT_EMAIL],
  };
};

export default payload;
