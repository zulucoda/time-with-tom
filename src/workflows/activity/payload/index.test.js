import payload from '.';
import activityConfig from '../constants';

describe('Workflows - Activity - Payload - Unit Test', () => {
  test('should return mapped payload values', () => {
    const values = {
      [activityConfig.SELECT_NAME]: activityConfig.ACTIVITY_LIST[1].value,
    };
    const act = payload(values);
    expect(act).toMatchSnapshot();
  });
});
