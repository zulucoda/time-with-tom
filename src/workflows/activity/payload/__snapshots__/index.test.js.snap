// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Workflows - Activity - Payload - Unit Test should return mapped payload values 1`] = `
Object {
  "activity": "golf",
}
`;
